bundles/scripting/core/src/main/java/org/apache/sling/scripting/core/impl/DefaultSlingScript.java

```diff
@@ -371,21 +377,31 @@ public void setReader(Reader reader) {
             }
 
             // evaluate the script
-            final Object result;
-            if (method == null && this.scriptEngine instanceof Compilable) {
-                CachedScript cachedScript = scriptCache.getScript(scriptName);
-                if (cachedScript == null) {
-                    ScriptNameAwareReader snReader = new ScriptNameAwareReader(reader, scriptName);
-                    CompiledScript compiledScript = ((Compilable) scriptEngine).compile(snReader);
-                    cachedScript = new CachedScriptImpl(scriptName, compiledScript);
-                    scriptCache.putScript(cachedScript);
-                    LOGGER.debug("Adding {} to the script cache.", scriptName);
+            Object result = null;
+            String selectorString = props.getRequest().getRequestPathInfo().getSelectorString();
+            if(StringUtils.isBlank(selectorString) || !selectorString.contains(BVP_VARIABLES_SELECTOR)) {
```

Comment:
what if someone defines a dedicated script to answer requests with selector "bvpvars". We should at least log a big warning in that case.